examples/LoggingExtensions/Program.cs

```diff
@@ -0,0 +1,51 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Logs;
+using OpenTelemetry.Resources;
+using Serilog;
+
+var resourceBuilder = ResourceBuilder.CreateDefault().AddService("Examples.LogEmitter");
+
+// Note: It is important that OpenTelemetryLoggerProvider is disposed when the
+// app is shutdown. In this example we allow Serilog to do that by calling CloseAndFlush.
+var openTelemetryLoggerProvider = new OpenTelemetryLoggerProvider(options =>
```

Comment:
Instead of assigning this to a variable `openTelemetryLoggerProvider` and explaining the object ownership/lifecycle, I suggest putting the `new OpenTelemetryLoggerProvider` expression directly in Ln:35 `WriteTo.OpenTelemetry(new OpenTelemetryLoggerProvider(...), disposeProvider: true)`: it increases the _locality_ when folks read the code, making the code more self-contained rather than highly contextual, it avoids accidental misuse (e.g. code holding a reference to disposed object).

Reply:
Here is the version from the other PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3305/files#diff-eebf3c37db933ef5eddad43dc03d2553b800b073573f291b7e0ac5999a1391c8

Where I am heading with this is two different types of extensions sharing a provider.

So the question is, what do we want to demo in the example? Sharing a provider or independent providers? I feel like shared will be the more common case. But LMK your thoughts on this.

Reply:
Here is my take:
1. `examples/LoggingExtensions/Program.cs` looks an example for the end-user - who is using the OpenTelemetry.Extensions.Serilog package to pipe Serilog via OpenTelemetry plumbing. For these users, normally they will only use Serilog in a single application, it makes more sense for them to have something simple, and not to worry about object lifecycle management except for the top level `CloseAndFlush`.
2. `examples/LogEmitter/Program.cs` is more targeting the plumbers, might make sense to expose more complexity / duty.

Reply:
@reyang Let me merge this and then I'll follow-up on this on my next PR where this example project is going to get more interesting?